### PR TITLE
GQI - Extensions - Logging - More detailed info and code example

### PR DESCRIPTION
**Pull request by skyline-rge**
On request, adding more concrete details and a concrete code example for [RN39043](https://intranet.skyline.be/DataMiner/Lists/Release%20Notes/DispForm2.aspx?ID=39043)

### CODE REVIEW

**Review by WaHaeck — APPROVED** [review]

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/GQI_Extensions_Logging.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+## Logging messages
+
+To log messages from within an ad hoc data source or custom operator:
+
+1.
+
+1.
+
+1.
+
```
[the remark this answers is not in the document]
Comment (skyline-rge) [reply user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/GQI_Extensions_Logging.md:26]: I think you might have misunderstood the original "How to log" section a bit? I'm sorry that it wasn't very clear.

The section you now labeled as "Prerequisites" contains in fact the exact steps you need to take in your GQI extension to enable logging. After that, creation a log message is just calling one of the IGQILogger methods as you would in any other program.

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/GQI_Extensions_Logging.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
+---
+uid: GQI_Extensions_Logging
+---
+
+# Logging
+
+From DataMiner 10.4.5/10.5.0 onwards<!-- RN 39043 -->, [ad hoc data sources](xref:Configuring_an_ad_hoc_data_source_in_a_query) and [custom operators](xref:GQI_Custom_Operator) can log their own messages and exceptions within GQI, complementing the existing [GQI core logging functionality](xref:GQI_Logging) (available from DataMiner 10.4.0/10.4.4).
+
+## Prerequisites
```
Comment (skyline-rge) [inline user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/GQI_Extensions_Logging.md:9]: I still like the original section title better here. Or you could replace it with your "Logging messages" as well.
Either way, this section contains the main steps I wanted to explain and not prerequisites.

```suggestion
## How to log
```

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/GQI_Extensions_Logging.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+To enable logging, the GQI extension (either an ad hoc data source or a custom operator) must implement the [IGQIOnInit](xref:GQI_IGQIOnInit) interface.
+
+This building block adds the [OnInit](xref:GQI_IGQIOnInit#oninitoutputargs-oninitoninitinputargs-args) lifecycle method, granting access to the [IGQILogger](xref:GQI_IGQILogger) object through the [Logger property](xref:GQI_OnInitInputArgs#properties) of the [OnInitInputArgs](xref:GQI_OnInitInputArgs) class.
+
+Throughout its lifespan, the extension uses the [IGQILogger](xref:GQI_IGQILogger) object to log messages and exceptions.
```
Comment (skyline-rge) [inline user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/GQI_Extensions_Logging.md:15]: The goal here was to stress how *you, the extension creator*,  can take action to log messages. Phrasing it like this gives the impression that there is more info needed to log your own messages but that's not the case.

```suggestion
Throughout its lifespan, the extension can use the [IGQILogger](xref:GQI_IGQILogger) methods to log messages and exceptions.
```

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/GQI_Extensions_Logging.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
+---
+uid: GQI_Extensions_Logging
+---
+
+# Logging
+
+From DataMiner 10.4.5/10.5.0 onwards<!-- RN 39043 -->, [ad hoc data sources](xref:Configuring_an_ad_hoc_data_source_in_a_query) and [custom operators](xref:GQI_Custom_Operator) can log their own messages and exceptions within GQI, complementing the existing [GQI core logging functionality](xref:GQI_Logging) (available from DataMiner 10.4.0/10.4.4).
+
+## How to log
```
Comment (MariekeGO) [inline user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/GQI_Extensions_Logging.md:9]: To be more in line with our usual phrasing, I'd recommend an -ing form here:

```suggestion
## Enabling logging
```

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/GQI_Extensions_Logging.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (EdithLansens) [reply user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/GQI_Extensions_Logging.md:16]: @skyline-rge, I fear my inexperience with GQI extensions is to blame for this - no worries! Is this PR OK to merge?

[the remark this answers is not in the document]
Comment (skyline-rge) [reply user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/GQI_Extensions_Logging.md:16]: @EdithLansens no blame, I'm glad you guys take the time to even try to understand everything :)
PR looks OK to me now.

**Review by EdithLansens — APPROVED** [review]

**Pull request merged**